### DISCUSSION TITLE
Sending Preview email not working with new template

### DISCUSSION THREAD

**Discussion by StickyPixels** [liedekef/events-made-easy#412] [Q&A] [answered]
Hi,

I have a weird issue with a new email template I designed.
Recently I adapted an existing email-template by adding a basic stylesheet and some minor tweaks, and I (Admin) can send preview emails using this new template. But my colleague (who has Editor permissions) can not.
She can still send preview emails when using the old template, but not when using the new template. Nothing happens when she clicks 'Send Preview Email', no warning, nothing.

The really weird thing is that when I log into her account (or any Editor account) from my computer, I can send previews without a problem using this template. So it's obviously not a permissions issue, all permissions are set to standard, email is handled by the default Wordpress email system.
I'm on a Mac M1, she is on Windows, but when she tries to send a preview from her iPhone (safari), it also refuses to send. So it may be an issue with her internet provider, but that should also apply to the old template as well, which works fine.

We are completely in the dark about this, and possibly it's not an EME issue but I can't think of anything, so I'm hoping someone with more knowledge about can shed some light.

Hope anyone can help, thanks in advance. 

I'm including the new template below, in case it contains any clues:

```
<html><head><style>@import url('https://fonts.googleapis.com/css2?family=Karla:ital,wght@0,400;0,700;1,400;1,700&family=Oswald&display=swap');body{background:#fff;color:#333;font-weight:normal;margin:0;padding:0;}a {color:#c00;text-decoration:none;transition:all .25s ease-in-out;}a:hover{color:#f00 !important;}img{max-width:100% !important; height:auto;}img.alignnone{margin:1em 0;}</style></head>
<body style="font-family: Karla, Verdana, sans-serif;">
<table border="0" cellspacing="0" cellpadding="0" style="width: 90%; max-width: 640px; margin: auto;">
<tbody>
<tr>
<td>
<div style="height: 130px; background-color: #555; background-image: url('https://tt-theater.nl/wp-content/images/stoeltjes.png');"><a title="TT Theaterproducties" href="https://tt-theater.nl/" rel="home"><img src="https://tt-theater.nl/wp-content/images/TT_logo_mail.png" /></a></div>
<h2 style="color: #fff; background-color:#c00; padding: 5px 15px; margin: 0; text-align: center;font-family: Oswald, Impact, sans-serif;font-weight: normal;">NIEUWSBRIEF</h2></td>
</tr>
<tr>
<td style="padding-top: 15px; padding-bottom: 30px;">

Beste #_FIRSTNAME,

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas iaculis tincidunt elit, vel euismod quam consectetur nec.

Praesent quis purus dignissim, aliquet lorem facilisis, mattis tellus. Etiam nec sapien eu lorem gravida dictum id id enim. Sed ac feugiat lorem. Donec eros libero, faucibus id ipsum sit amet, elementum volutpat enim. 
<h2 style="color: #fff; background-color:#c00; padding: 5px 15px;clear:both;font-family: Oswald, Impact, sans-serif;font-weight: normal;">Quisque vulputate</h2>
Quisque vulputate accumsan libero, sed maximus nisl commodo in. Nulla gravida nulla ac erat posuere, sed laoreet felis tempor. Nam id fringilla mi. In eleifend tempor lacus, laoreet consequat nulla molestie sit amet.
<h2 style="color: #c00;clear:both;font-family: Oswald, Impact, sans-serif;font-weight: normal;">Vivamus mauris lacus</h2>
Vivamus mauris lacus, interdum vel nibh vel, dignissim aliquam magna. Nullam iaculis, tortor ac vehicula euismod, erat odio vehicula massa, sit amet pellentesque nisi odio eu quam. Curabitur nec maximus nibh. Proin hendrerit tortor eu augue suscipit, vitae condimentum augue lobortis. Suspendisse quis sem eu risus mollis eleifend non ut massa. Morbi tincidunt ex vitae nisi ullamcorper suscipit nec ut lorem.</td>
</tr>
<tr>
<td style="color: #fff; background-color: #c00; padding: 15px 0; text-align: center; font-size: .8em;">Deze e-mail is verzonden aan <a style="color: #fff; text-decoration: underline;" href="mailto:#_EMAIL">#_EMAIL</a><br>
Als je geen nieuwsbrief meer wilt ontvangen, kun je je <a style="color: #fff; text-decoration: underline;" href="https://tt-theater.nl/over-tt/contact/afmelden-nieuwsbrief/">hier afmelden</a>.</td>
</tr>
<tr>
<td style="text-align: center; background-color: #555; background-image: url('https://tt-theater.nl/wp-content/images/stoeltjes.png');">
<table border="0" cellspacing="0" cellpadding="10" style="margin: auto; font-size: .8em;">
<tbody>
<tr>
<td>
<div style="color: #fff;" align="center"><a href="https://www.facebook.com/tttheater/"><img src="https://tt-theater.nl/wp-content/images/FB-logo.png" width="30" height="30" /></a>

facebook</div></td>
<td>
<div style="color: #fff;" align="center"><a href="https://tt-theater.nl/"><img src="https://tt-theater.nl/wp-content/images/TT-site.png" width="30" height="30" /></a>

website</div></td>
</tr>
</tbody>
</table>
</td>
</tr>
</tbody>
</table>
</body></html>
```

**Comment by liedekef** [comment]
It is certainly not an internet provider issue, mails are sent via EME (so the client has no impact there). However, it may be that she has a javascript error in her browser which might (will) prevent mails from being sent (the form submission is done via javascript/jquery and if an error is encountered on the page, this no longer works). So best check her browser console for hints.

**Comment by StickyPixels** [comment]
Thank you very much, I'll take a look into her console. The possibility already crossed my mind but I dismissed it because the problem exists in two different browsers and with only one template. But it really is the only possibility that makes sense.
I'll keep you posted, thanks a bunch.

**Comment by StickyPixels** [comment, marked as answer]
Hi Franky,
I finally solved the riddle, it turns out the mail action was blocked by our firewall.
I use WordFence on our site for protection, and in the weekly report there were warnings for XSS blocking of mail messages. In WF dashboard I found this message for all earlier attempts:
`https://sitename.url/wp-admin/admin.php?page=eme-emails and was blocked by firewall for XSS: Cross Site Scripting in POST body: generic_mail_message=[etc]`
I was able to send previews because my IP address was in the exceptions list, so I added my colleagues IP and now it works for her as well.
Thank you for your time, happy it's solved.

**Comment by liedekef** [comment]
A bit weird that wordfence sees it as a XSS though ... calls to admin.php are normal when logged in, so if you ever figure out why this is, maybe I can work around it in a future version.

**Comment by StickyPixels** [comment]
I'm guessing the issue has to do with the stylesheet I added to the email header. Templates without stylesheet worked fine all the time, but when I added a styleheet the problem occurred. Apparently Wordfence considers a stylesheet to be a 'script'?
Hope this helps clearing it up.

**Reply by liedekef** [reply]
This post might be of help then: https://wordpress.org/support/topic/google-tag-manager-blocked-for-xss/
